resource/environment/integration.js

```diff
@@ -1,49 +1,14 @@
+const _ = require( "lodash" );
+const Promise = require("bluebird");
```

Comment:
@arobson Looks like bluebird was brought in to apply the generator based co-routine pattern. It looks like when.js supports this functionality with ```when/generator```.

In my experience, Bluebird gives better stack traces than when.js and I believe is more performant. The APIs are pretty similar. I agree though, it's best to stick to one library.